### PR TITLE
Configurations/10-main.conf: add support for -arch x86_64h on macOS

### PR DESCRIPTION
**Pull request by MarcusCalhoun-Lopez**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by dot-asm** [comment]
Separate x86_64h target doesn't make much sense. Just in case for reference, `h` apparently stands for Haswell, and supposedly it allows compiler to generate AVX2 code. For starters, compiling OpenSSL **itself** with -arch x86_64h doesn't give any significant benefit, because most critical code paths are implemented in assembly, and remaining compiler-generated AVX2 wouldn't make much difference [if any at all]. So that the only possible reason to compile OpenSSL with -arch x86_64h would be when 3rd-party application code is compiled with -arch x86_64h. However!!! As far as shared libraries go, application compiled with -arch x86_64h is perfectly capable of linking with shared library compiled with -arch x86_64. And as already mentioned, in OpenSSL case it wouldn't make difference performance-wise, so there is no real need to compile OpenSSL shared libraries with -arch x86_64h. It's only problematic to link with **static** libraries. I mean if one compiles one module with -arch x86_64h, then all object files, including those coming from static libraries, have to be of same processor sub-type. And here is question. Is it possible to solve the problem with default config? And the answer is it totally is possible. If you pass additional `-arch%20x86_64h` argument at config-time, it will generate fat static libraries that would be right for linking with application code compiled with either x86_64h or x86_64.

To summarize. For shared build, just use default. For static build, just pass additional `-arch%20x86_64h` at config-time (if x86_64h actually helps *your* application).

**Comment by richsalz** [comment]
So closing this.

**Comment by dot-asm** [comment]
There is one related thing tough. The fact that OpenSSL is compiled with -arch x86_64 does **not** prevent binary from executing AVX2 (or better) assembly code paths. But whether or not those code paths are emitted depends on perlasm scripts detecting clang version to determine if it can actually assemble them. And since Apple seems to change version output string (again?), version is not detected and AVX2 code paths are not emitted. Just in case, no, passing -arch x86_64h would not make perlasm detect version. So that this remark is not in favour of passing x86_64h. Everything that was said in initial remark still stands.

**Comment by dot-asm** [comment]
> Apple seems to change version output string (again?)

As it turns out it's not like they changed it "again", but rather that the only file I checked this time, rsaz-avx2, was left behind when last change was recognized. Few 32-bit ones were left behind too. It's being addressed in #6499.

**Comment by MarcusCalhoun-Lopez** [comment]
@dot-asm, thank you for your analysis and explanation.
They were very clear and point a way forward on my end.
